src/test/java/com/google/sps/TripTest.java

```diff
@@ -0,0 +1,94 @@
+
+package com.google.sps;
+
+import com.google.sps.Trip;
+import com.google.sps.TripDay;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+
+@RunWith(JUnit4.class)
+public final class TripTest {
+  private static final String TRIP_NAME = "Trip to California";
+  private static final String START_DATE = "2020-06-25";
+  private static final String END_DATE = "2020-06-28";
+  private static final int NUM_DAYS = 4;
+  
+  private ArrayList<TripDay> tripDays;
+
+  private static final String TIMES_SQUARE_ID = "ChIJmQJIxlVYwokRLgeuocVOGVU";
+  private static final String CENTRAL_PARK_ID = "ChIJ4zGFAZpYwokRGUGph3Mf37k";
+  private static final String WORLD_TRADE_ID = "ChIJy7cGfBlawokR5l2e93hsoEA";
+  private static final String EMPIRE_STATE_ID = "ChIJtcaxrqlZwokRfwmmibzPsTU";
+  private static final String HOTEL_ID = "ChIJ68J3tfpYwokR2HaRoBcB4xg";
+
+  private ArrayList<String> locations = new ArrayList<>(Arrays.asList(TIMES_SQUARE_ID, CENTRAL_PARK_ID,
+                                                                    WORLD_TRADE_ID, EMPIRE_STATE_ID));
+
+  // Test the trip constructor which allows for multiple days
+  @Test
+  public void testTripConstructorMultiDay() {
+    TripDay tripDay1 = new TripDay(HOTEL_ID, HOTEL_ID, locations);
+    TripDay tripDay2 = new TripDay(HOTEL_ID, HOTEL_ID, locations);
+    
+    tripDays = new ArrayList<>();
+    tripDays.add(tripDay1);
+    tripDays.add(tripDay2);
+
+    Trip trip = new Trip(TRIP_NAME, START_DATE, END_DATE, NUM_DAYS, tripDays);
+
+    Assert.assertEquals(trip.getTripName(), TRIP_NAME);
+    Assert.assertEquals(trip.getStartDate(), START_DATE);
+    Assert.assertEquals(trip.getEndDate(), END_DATE);
+    Assert.assertEquals(trip.getNumDays(), NUM_DAYS);
+    Assert.assertEquals(trip.getTripDays(), tripDays);
+  }
+
+  // Test the Trip constructor for a single day (MVP version)
+  @Test
+  public void testTripConstructorSingleDay() {
+    TripDay tripDay = new TripDay(HOTEL_ID, HOTEL_ID, locations);
+    
+    tripDays = new ArrayList<>();
+    tripDays.add(tripDay);
+
+    Trip trip = new Trip(TRIP_NAME, START_DATE, tripDays);
+
+    Assert.assertEquals(trip.getTripName(), TRIP_NAME);
+    Assert.assertEquals(trip.getStartDate(), START_DATE);
+    Assert.assertEquals(trip.getEndDate(), START_DATE);
+    Assert.assertEquals(trip.getNumDays(), 1);
+    Assert.assertEquals(trip.getTripDays(), tripDays);
+    
+  }
+
+  // Test multiple day constructor with null parameter
+  @Test(expected = IllegalArgumentException.class)
+  public void testTripConstructorMultiDayNull() {
+    Trip trip = new Trip(TRIP_NAME, START_DATE, END_DATE, NUM_DAYS, null);
+  }
+
+  // Test single day constructor with null parameter
+  @Test(expected = IllegalArgumentException.class)
+  public void testTripConstructorSingleDayNull() {
+    Trip trip = new Trip(TRIP_NAME, START_DATE, null);
+  }
```

Comment:
If you end up adding code to verify that the dates are valid and that they match the other inputs, can we also add tests checking this functionality as well? (Only applicable if you end up adding that initial code, of course.)